### PR TITLE
IMGMOUNT command to display current disk image status

### PR DESCRIPTION
**Pull request by Wengier**
This PR allows IMGMOUNT command (without parameters) to display the current status of disk and CD images, including the current disk names, labels, and swap slots (e.g. 2 / 3). The format is somewhat similar to MOUNT command without parameters, but with different information displayed to users. The PR helps solve issue #997.

### CODE REVIEW

File: src/dos/drives.cpp
Comment (kcgen) [inline src/dos/drives.cpp:262]: Is it possible to use `safe_sprintf`?

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:56]: Is the `11` related to `"%-*s %-*s %-*s %-*s\n"`? 
Maybe mention something about the 11 being the minimum allowable width(if that's what it is).

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:51]: If some of these widths are coming in from the messaging system, then they might be written by translators, and therefore are truly "user-side content".

My concerns is that, if some of those messages are quite wide, the subtractions could take width_2 into negative territory. 

I'm not sure if there's a hard-minimum or if width_2 is allowed to go negative? (maybe it's OK?), but if not, a runtime check could help.

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:62]: Haven't checked this, but it might do:
```c++
const std::string horizontal_divider = std::string s(term_width, '-'); 
WriteOut_NoParsing(horizontal_divider.c_str());
```

File: src/dos/program_imgmount.h
Comment (kcgen) [inline src/dos/program_imgmount.h:28]: very minor style change.. we always knockout `(void)` for `()` when possible. 
`void ListImgMounts();`, because it matches how the caller uses it, and is a bit easy to read.

File: src/dos/drives.cpp
Comment (Wengier) [reply src/dos/drives.cpp]: Right, it should be used instead. Done.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: The 11 is based on the MOUNT code, and now I define it as a const int named "minimum_length" which is common to both IMGMOUNT and MOUNT commands.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: I added a run-time check as suggested. Done.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: Done for both IMGMOUNT and MOUNT commands.

File: src/dos/program_imgmount.h
Comment (Wengier) [reply src/dos/program_imgmount.h]: Done.

**Comment by Wengier** [comment]
@kcgen Thanks for the review. I have addressed the points mentioned above, applying to both IMGMOUNT and MOUNT commands where applicable.

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:51]: Suggest using string's `.length()` instead of `.size()`.
Length also makes it read more consistent when comparing against `minimum_length`.

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:63]: I'm not sure if these colums have any different meanings from each other?
when I see `txt_1`, `txt_2`, `txt_3`, ... it tells me we're dealing with ambiguous "bulk data".

But if these columns have some meaning, lets hold the data in variables named after those meanings.

File: src/dos/program_imgmount.cpp
Comment (kcgen) [inline src/dos/program_imgmount.cpp:70]: Suggest a name that impies a yes or no state or tense.
"print drive", to me, means the drive to print - and it would hold whatever a "drive" is.
Adding words like: is, has, did, could, maybe, should, would, have, ... all implies a yes-or-no state.
In this case, maybe `found_drives` would do? ("found" also implies a yes/no state).

File: src/dos/program_mount.cpp
Comment (kcgen) [inline src/dos/program_mount.cpp:112]: Oh - I think we can get one simpler:
``` c++
const std::string horizontal_divider(term_width, '-');
```

File: src/dos/program_mount_common.h
Comment (kcgen) [inline src/dos/program_mount_common.h:34]: With this floating on its own in the header, it feels a bit disconnected ("length" in this case might mean something about the drives being mounted, as opposed to column widths).  

Suggest adding a comment or naming it more specifically to explain/describe that this is a column width minimum length? (or maybe it's not a column width.. I don't actually know.. but mentioning it as an example).

**Review by kcgen — COMMENTED** [review]
Round 2 done -just some minor points remaining. Thanks for this improvement @Wengier !

File: src/dos/program_mount.cpp
Comment (Wengier) [reply src/dos/program_mount.cpp]: Nice simplification. Done.

File: src/dos/program_mount_common.h
Comment (Wengier) [reply src/dos/program_mount_common.h]: The header applies to both IMGMOUNT and MOUNT commands, which was why I moved it here. Now I have changed the name to be more descriptive and also added a comment.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: Changed to found_drives as suggested. Done.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: Changed such names to more descriptive ones for both IMGMOUNT and MOUNT commands. Done.

File: src/dos/program_imgmount.cpp
Comment (Wengier) [reply src/dos/program_imgmount.cpp]: Done.

**Comment by kcgen** [comment]
Looks great @Wengier - I can't see anything else!

Do you think the `Address issues as suggested` commit could be squashed with the prior, so this PR is just one commit?

(this is just a question for your assessment: if you think it would help to bisect to this "reviewed-changes" versus "original changes", then yes - we should keep both commits).  But if the changes are topical, then we can squash.  your call :+1:

**Review by kcgen — APPROVED** [review]
(Approved - just one question regarding squashing before we merge)

**Pull request merged**
